cfme/fixtures/vm.py

```diff
@@ -9,13 +13,68 @@
 from cfme.utils.wait import wait_for
 
 
+DOCSTRING_CACHE = {}
+TEST_PARAMS = re.compile(r"\[.*\]")
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        '--no-assignee-vm-name',
+        action='store_true',
+        default=False,
+        help=(
+            'If passed no-assignee-vm-name, vm names will not contain test_case assignee names.'
+        )
+    )
+
+
+@pytest.hookimpl(trylast=True)
+def pytest_collection_modifyitems(session, config, items):
+    if not config.getoption('--no-assignee-vm-name'):
+        process_json_data(session, items, all_items=True)
+
+
 def _create_vm(request, template, provider, vm_name):
+    if not pytest.config.getoption('--no-assignee-vm-name'):
+        if isinstance(request.node, pytest.Function):
+            nodeid = TEST_PARAMS.sub('', get_nodeid_full_path(request.node))
+            assignee = request.session._docstrings_cache[nodeid].get('assignee', '')
+        else:
+            # Fetch list of tests in the module object
+            test_list = [
+                item
+                for item in dir(request.module)
+                if ('test_' in item[:5]) and not ('test_requirements' == item)
```

Comment:
Can you add a comment why you are taking items 0-5?

Reply:
Looks like this is just an odd way of expressing `item.startswith('test_')`